x-pack/solutions/observability/plugins/observability_agent/server/tools/get_change_point/index.ts

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { omit, orderBy } from 'lodash';
+import { z } from '@kbn/zod';
+import { ToolType } from '@kbn/onechat-common';
+import { ToolResultType } from '@kbn/onechat-common/tools/tool_result';
+import type { BuiltinToolDefinition } from '@kbn/onechat-server';
+import type { CoreSetup, Logger } from '@kbn/core/server';
+import { SavedObjectsClient } from '@kbn/core/server';
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import type { AggregationsAutoDateHistogramAggregation } from '@elastic/elasticsearch/lib/api/types';
+import type {
+  ObservabilityAgentPluginSetupDependencies,
+  ObservabilityAgentPluginStart,
+  ObservabilityAgentPluginStartDependencies,
+} from '../../types';
+import { getMetricChangePoint } from './get_metric_change_point';
+import { getLogChangePoint } from './get_log_change_point';
+
+export const OBSERVABILITY_GET_LOG_CHANGE_POINT_TOOL_ID = 'observability.get_log_change_point';
+export const OBSERVABILITY_GET_METRIC_CHANGE_POINT_TOOL_ID =
+  'observability.get_metric_change_point';
+
+const dateHistogram: AggregationsAutoDateHistogramAggregation = {
+  field: '@timestamp',
+  buckets: 100,
+};
+
+const getFilters = ({
+  start,
+  end,
+  kqlFilter,
+}: {
+  start: string;
+  end: string;
+  kqlFilter?: string;
+}) => {
+  const filters = [
+    {
+      range: {
+        '@timestamp': {
+          gte: start,
+          lt: end,
+        },
+      },
+    },
+  ];
+  return [...filters, ...(kqlFilter ? [toElasticsearchQuery(fromKueryExpression(kqlFilter))] : [])];
+};
+
+const getLogChangesSchema = z.object({
```

Comment:
To match the tool name:
```suggestion
const getLogChangePointsSchema = z.object({
```